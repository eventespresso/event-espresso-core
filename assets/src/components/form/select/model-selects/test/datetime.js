import DatetimeSelect, { EditorDatetimeSelect } from '../datetime';
import { shallow, render } from 'enzyme';

jest.mock( '../../model-select', () => () => <span>DatetimeSelect</span> );

describe( 'DatetimeSelect()', () => {
	const expectedDefaultQueryData = {
		limit: 100,
		order: 'DESC',
		orderBy: 'start_date',
		showExpired: false,
	};
	it( 'matches snapshot with default props', () => {
		const wrapper = shallow( <DatetimeSelect /> );
		expect( wrapper ).toMatchSnapshot();
	} );
	it( 'matches expected default props', () => {
		const wrapper = shallow( <DatetimeSelect /> );
		expect( wrapper.prop( 'queryData' ) )
			.toEqual( expectedDefaultQueryData );
		expect( wrapper.state( 'queryData' ) )
			.toEqual( expectedDefaultQueryData );
		expect( wrapper
			.prop( 'selectConfiguration' ).placeholder ).toEqual(
			'Select Datetime...',
		);
		expect( wrapper.prop( 'modelName' ) ).toEqual( 'datetime' );
		expect( wrapper.prop( 'label' ) ).toEqual( 'Select Datetime' );
	} );
	it( 'matches expected update to queryData state with forEventId prop',
		() => {
			const wrapper = shallow( <DatetimeSelect forEventId={ 20 } /> );
			expect( wrapper.state( 'queryData' ) ).toEqual( {
				...expectedDefaultQueryData,
				forEventId: 20,
			} );
		},
	);
} );

describe( 'EditorDatetimeSelect()', () => {
	it( 'matches snapshot with default props', () => {
		const wrapper = render( <EditorDatetimeSelect /> );
		expect( wrapper ).toMatchSnapshot();
	} );
} );
