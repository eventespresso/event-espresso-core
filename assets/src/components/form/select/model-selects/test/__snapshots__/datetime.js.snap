// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatetimeSelect() matches snapshot with default props 1`] = `
<Component
  getQueryString={[Function]}
  modelName="datetime"
  queryData={
    Object {
      "limit": 100,
      "order": "DESC",
      "orderBy": "start_date",
      "showExpired": false,
    }
  }
  selectConfiguration={
    Object {
      "defaultValue": undefined,
      "loadingMessage": [Function],
      "noOptionsMessage": [Function],
      "onChange": undefined,
      "placeholder": "Select Datetime...",
    }
  }
  selectLabel="Select Datetime"
/>
`;

exports[`EditorDatetimeSelect() matches snapshot with default props 1`] = `
<div
  class="components-base-control"
>
  <div
    class="components-base-control__field"
  >
    <span>
      DatetimeSelect
    </span>
  </div>
</div>
`;
